modules/core/arrow-data/src/test/kotlin/arrow/data/OptionTTest.kt

```diff
@@ -23,68 +21,56 @@ class OptionTTest : UnitSpec() {
         a.value() == b.value()
     }
 
-    val NELM: Monad<ForNonEmptyList> = monad<ForNonEmptyList>()
+    val NELM: Monad<ForNonEmptyList> = NonEmptyList.monad()
 
     init {
 
-        "instances can be resolved implicitly" {
-            functor<OptionTPartialOf<ForNonEmptyList>>() shouldNotBe null
-            applicative<OptionTPartialOf<ForNonEmptyList>>() shouldNotBe null
-            monad<OptionTPartialOf<ForNonEmptyList>>() shouldNotBe null
-            foldable<OptionTPartialOf<ForNonEmptyList>>() shouldNotBe null
-            traverse<OptionTPartialOf<ForNonEmptyList>>() shouldNotBe null
-            semigroupK<OptionTPartialOf<ForListK>>() shouldNotBe null
-            monoidK<OptionTPartialOf<ForListK>>() shouldNotBe null
-            functorFilter<OptionTPartialOf<ForListK>>() shouldNotBe null
-            traverseFilter<OptionTPartialOf<ForOption>>() shouldNotBe null
-        }
-
         testLaws(
-            MonadLaws.laws(OptionT.monad(NonEmptyList.monad()), Eq.any()),
-            SemigroupKLaws.laws(
-                OptionT.semigroupK(Id.monad()),
-                OptionT.applicative(Id.monad()),
-                EQ()),
-
-            MonoidKLaws.laws(
-                OptionT.monoidK(Id.monad()),
-                OptionT.applicative(Id.monad()),
-                EQ()),
-
-            FunctorFilterLaws.laws(
-                OptionT.functorFilter(),
-                { OptionT(Id(Some(it))) },
-                EQ()),
-
-            TraverseFilterLaws.laws(
-                OptionT.traverseFilter(),
-                OptionT.applicative(Option.monad()),
-                { OptionT(Option(Some(it))) },
-                EQ(),
-                EQ_NESTED())
+                MonadLaws.laws(OptionT.monad(NonEmptyList.monad()), Eq.any()),
+                SemigroupKLaws.laws(
+                        OptionT.semigroupK(Id.monad()),
+                        OptionT.applicative(Id.monad()),
+                        EQ()),
+
+                MonoidKLaws.laws(
```

Comment:
Should these be declared as extensions functions too in order to be coherent with the style?

Reply:
No, at some point the style requires changing back to dynamic dispatch IMO. Fixable at a later point anyway.